_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+| **merge** feature branch to `master`                  | Release on `latest` channel if the commit trigger an allowed release type, report an error otherwise                                                                                                                                                                                                      |
+| **push** to `next`                                    | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** any branch to `next`                        | Release on `next` channel                                                                                                                                                                                                                                                                                 |
+| **merge** some commits from `next` branch to `master` | Get the last release for commits on branch `master` (including the one that come from `next`), analyze commits, increase the version; If the commit associated with the last release found is also present on `next`, make the version available on the `latest` channel, otherwise do a regular release. |
+
+##### Future branches configuration
+
+```json
+{
+  "release": {
+    "branches": ["master", "next"]
+  }
+}
+```
+
+Internally **semantic-release** will determines the type of commit that can be done on each branch, based on the number of future branches defined:
+- `"branches": ["master"]` => Any type of commit can be pushed to `master`
+- `"branches": ["master", "next"]` => `master` will accept `fix` and `feat` commits; `next` will accept any type of commits.
+- `"branches": ["master", "next", "next-major"]` => `master` will accept `fix`; `next` will accept `fix` and `feat` commits; `next-major` will accept any type of commits.
+
+Those rules are meant to prevent inconsistent state in releases in which the same releases line is "forked".
+
+For example with the config `"branches": ["master", "next"]` and:
+- last release from `master` is `1.1.0`
+- last release from `next` is `2.0.0`
+
+Committing a breaking change to `master` would trigger the release of `2.0.0` that would fail as the release already exists.
+
+In addition when the `master` and `next` have the same commit head (after a merging `next` into `master` for example), the first commit done on `next` must be a breaking change.
```

Comment:
Does this mean that I could never release a new feature on `next`? All new features would be deployed straight to `latest`?

Reply:
If you want to release new feature on `@next` you would need a 3 branch setup, for example:
- `master` => fixes on `@latest`
- `next` => features and subsequent fixes and  on `@next`
- `next-major` => breaking and subsequent features and fixes on `@next-major`

A two branch setting would look like this:
- `master` => fixes and features on `@latest`
- `next` => breaking and subsequent features and fixes on `@next-major`

Reply:
To clarify a bit further, in a two branch setting, yes you can deploy a new feature on `@next`, as long as the `next` branch is at least one breaking change ahead of `master`.

The comment is limited to situation where
> the `master` and `next` have the same commit head (after a merging `next` into `master` for example)

Reply:
Okay, so releasing enhancements to early adopters through a `next` dist-tag _is_ possible using the _3 branch_ approach you described above.

There are cases where a new, non-breaking, enhancement, contains significant changes to code such that I would like to deploy that version to a set of early adopters, prior to deploying it for general consumption.

Reply:
Everything is based on the commit convention and the semantic release type. So there is now to distinguish a feature that's suppose to go to `@latest` from a feature that's big enough to go to `@next`.